test/integration_test.go

```diff
@@ -4993,9 +5002,91 @@ func (ts *IntegrationTestSuite) TestScheduleUpdate() {
 	})
 	ts.NoError(err)
 
-	description2, err := handle.Describe(ctx)
+	ts.Eventually(func() bool {
+		description2, err := handle.Describe(ctx)
+		ts.NoError(err)
+		ts.Equal(description.Schedule, description2.Schedule)
+		ts.Equal(2, description2.TypedSearchAttributes.Size())
+		returnedSa, _ := description2.TypedSearchAttributes.GetString(stringKey)
+		expectedSa, _ := sa.GetString(stringKey)
+		ts.Equal(expectedSa, returnedSa)
+		returnedSa, _ = description2.TypedSearchAttributes.GetKeyword(keywordKey)
+		expectedSa, _ = sa.GetKeyword(keywordKey)
+		ts.Equal(expectedSa, returnedSa)
+		return len(description2.SearchAttributes.IndexedFields) == 2
+	}, time.Second, 100*time.Millisecond)
+
+	// nil search attributes should leave current search attributes untouched
+	updateFunc = func(input client.ScheduleUpdateInput) (*client.ScheduleUpdate, error) {
+		return &client.ScheduleUpdate{
+			Schedule: &input.Description.Schedule,
+		}, nil
+	}
+
+	err = handle.Update(ctx, client.ScheduleUpdateOptions{
+		DoUpdate: updateFunc,
+	})
 	ts.NoError(err)
-	ts.Equal(description.Schedule, description2.Schedule)
+
+	ts.Eventually(func() bool {
+		description2, err := handle.Describe(ctx)
+		ts.NoError(err)
+		ts.Equal(2, description2.TypedSearchAttributes.Size())
+		returnedSa, _ := description2.TypedSearchAttributes.GetString(stringKey)
+		expectedSa, _ := sa.GetString(stringKey)
+		ts.Equal(expectedSa, returnedSa)
+		returnedSa, _ = description2.TypedSearchAttributes.GetKeyword(keywordKey)
+		expectedSa, _ = sa.GetKeyword(keywordKey)
+		ts.Equal(expectedSa, returnedSa)
+		return len(description2.SearchAttributes.IndexedFields) == 2
+	}, time.Second, 100*time.Millisecond)
+
+	// updating a single search attribute on an existing collection acts as an upsert on the entire collection
+	newSa := temporal.NewSearchAttributes(stringKey.ValueSet("Changed"))
+	updateFunc = func(input client.ScheduleUpdateInput) (*client.ScheduleUpdate, error) {
+		return &client.ScheduleUpdate{
+			Schedule:              &input.Description.Schedule,
+			TypedSearchAttributes: &newSa,
+		}, nil
+	}
```

Comment:
I think it's worth demonstrating/testing what updating an attribute without affecting the others looks like. I think it'd be something like:

```go
	updateFunc = func(input client.ScheduleUpdateInput) (*client.ScheduleUpdate, error) {
		newSa := temporal.NewSearchAttributes(
			input.TypedSearchAttributes.Copy(),
			stringKey.ValueSet("Changed"),
		)
		return &client.ScheduleUpdate{
			Schedule:              &input.Description.Schedule,
			TypedSearchAttributes: &newSa,
		}, nil
	}
```

Then check that it only updated that one.